src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -261,7 +261,7 @@ def load_arguments(self, _):
         c.argument('nsg', help='The name to use when creating a new Network Security Group (default) or referencing an existing one. Can also reference an existing NSG by ID or specify "" for none.', arg_group='Network')
         c.argument('nsg_rule', help='NSG rule to create when creating a new NSG. Defaults to open ports for allowing RDP on Windows and allowing SSH on Linux.', arg_group='Network', arg_type=get_enum_type(['RDP', 'SSH']))
         c.argument('application_security_groups', min_api='2017-09-01', nargs='+', options_list=['--asgs'], help='Space-separated list of existing application security groups to associate with the VM.', arg_group='Network')
-
+        c.argument('workspace', arg_group='Monitor', help='Name or ID of Log Analytics Workspace. If name is given, the workspace should be in the same resource group with the vm, otherwise a new workspace will be created.')
```

Comment:
Do you mean "name or ID is given"? If yes, "If the parameter/argument is given" may be better??

Reply:
no. If user give ID, it will try to connect the workspace. If user give name, it will check existence under same resource group. otherwise, create a new one.

Reply:
In this way, it will be more clear to use "If you specify the workspace through its name, your workspace should be in the same resource group with vm...." Just for suggestion, you can change it as you like or keep your existing one.